django/bossspatialdb/views.py

```diff
@@ -453,149 +412,7 @@ def post(self, request, collection, experiment, channel):
-            frame = request.data
-        else:
-            frame = {}
-
-        # Call Step Function
-        boss_config = bossutils.configuration.BossConfig()
-        experiment = resource.get_experiment()
-        coord_frame = resource.get_coord_frame()
-        lookup_key = resource.get_lookup_key()
-        col_id, exp_id, ch_id = lookup_key.split("&")
-
-        def get_frame(idx):
-            return int(frame.get(idx, getattr(coord_frame, idx)))
-
-        args = {
-            'collection_id': int(col_id),
-            'experiment_id': int(exp_id),
-            'channel_id': int(ch_id),
-            'annotation_channel': not channel.is_image(),
-            'data_type': resource.get_data_type(),
-
-            's3_bucket': boss_config["aws"]["cuboid_bucket"],
-            's3_index': boss_config["aws"]["s3-index-table"],
-
-            'x_start': get_frame('x_start'),
-            'y_start': get_frame('y_start'),
-            'z_start': get_frame('z_start'),
-
-            'x_stop': get_frame('x_stop'),
-            'y_stop': get_frame('y_stop'),
-            'z_stop': get_frame('z_stop'),
-
-            'resolution': int(channel.base_resolution),
-            'resolution_max': int(experiment.num_hierarchy_levels),
-            'res_lt_max': int(channel.base_resolution) + 1 < int(experiment.num_hierarchy_levels),
-
-            'type': experiment.hierarchy_method,
-            'iso_resolution': int(resource.get_isotropic_level()),
-
-            # This step function executes: boss-tools/activities/resolution_hierarchy.py
-            'downsample_volume_lambda': boss_config['lambda']['downsample_volume'],
-
-            'aws_region': get_region(),
-
-        }
-
-        # Check that only administrators are triggering extra large downsamples
-        if (not request.user.is_staff) and \
-           ((args['x_stop'] - args['x_start']) * \
-            (args['y_stop'] - args['y_start']) * \
-            (args['z_stop'] - args['z_start']) > settings.DOWNSAMPLE_MAX_SIZE):
-            return BossHTTPError("Large downsamples require admin permissions to trigger. Invalid Request.", ErrorCodes.INVALID_STATE)
-
-        # Add metrics to CloudWatch
-        def get_cubes(axis, dim):
-            extent = args['{}_stop'.format(axis)] - args['{}_start'.format(axis)]
-            return -(-extent // dim) ## ceil div
-
-        cost = (  get_cubes('x', 512)
-                * get_cubes('y', 512)
-                * get_cubes('z', 16)
-                / 4 # Number of cubes for a downsampled volume
- #               * 0.75 # Assume the frame is only 75% filled
-#                * 2 # 1 for invoking a lambda
-#                    # 1 for time it takes lambda to run
-#                * 1.33 # Add 33% overhead for all other non-base resolution downsamples
-               )
-
-        log = bossLogger()
-        log.info("Checking for throttling")
-        BossThrottle().check('downsample','compute',
```

Comment:
Where did the throttle check code get placed? I didn't see it in the downsample changes

Reply:
To follow up from the standup, I accidentally lost this, but we also discussed that you would improve the throttling logic in your branch.